### PR TITLE
Improve CI caching, and give each job names

### PR DESCRIPTION
**Pull request by madsmtm**
Two things to discuss:
1. The move from `hecrj/setup-rust-action` to `dtolnay/rust-toolchain` (done because of slightly better performance, and it seems like this is what everyone is moving to) removes support for the problem matchers, see https://github.com/dtolnay/rust-toolchain/issues/50. We can [re-add this explicitly](https://github.com/r7kamura/rust-problem-matchers) if desired, so: Is it desired? Do people actually like this feature? I myself don't ever use it, as I find it is really cluttered anyhow (since it's run on so many platforms, and GitHub doesn't do any deduplication).
2. Would it be time to start committing the `Cargo.lock`? Because if we do, we can use [Swatinem/rust-cache](https://github.com/Swatinem/rust-cache) or do something similarly easier instead. This should also prevent unrelated updates breaking CI in the future.

### CODE REVIEW

**Comment by madsmtm** [comment]
I'd estimate the build got on average 2 minutes faster (10 mins -> 8 mins) (depending on time of day and amount of concurrent builds)

**Review by kchibisov — CHANGES_REQUESTED** [review]
The changes are not equivalent to what we had before, in particular, features are not respected (I don't see them on CI jobs for example). Verbose was removed in some places. And the CI itself is much more complicated now, though, if it's faster than it's fine...

File: .github/workflows/ci.yml
Comment (madsmtm) [reply .github/workflows/ci.yml:103]: I... Intended to fold that into `$OPTIONS`, but it seems I made a mistake along the way

File: .github/workflows/ci.yml
Comment (madsmtm) [reply .github/workflows/ci.yml:18]: I'm pretty sure the `--all` argument is redundant, that's what happens by default? But let me check first

Comment (kchibisov) [reply .github/workflows/ci.yml:18]: I don't think it formats examples?

File: .github/workflows/ci.yml
Comment (madsmtm) [reply .github/workflows/ci.yml:103]: Nvmd, this works as intended; `$OPTIONS` now contains the features as well

File: .github/workflows/ci.yml
Comment (madsmtm) [reply .github/workflows/ci.yml:18]: Documentation for `--all`:
> Format all packages, and also their local path-based dependencies

But from a quick test `cargo fmt` by itself formats examples and the `run-wasm` crate.

**Comment by madsmtm** [comment]
The features are folded into `$OPTIONS`, and the `CARGO_TERM_VERBOSE: true` makes everything run in verbose mode.

I do agree about the complexity, unfortunate but difficult to do otherwise without committing `Cargo.lock`.

**Comment by madsmtm** [comment]
I believe I've resolved the outstanding concerns, so I'll merge this fairly soon unless there are others?

**Comment by kchibisov** [comment]
@madsmtm what's the reason CI is not being fully run? I don't want to merge a patch that changes CI which can't run everything on the CI itself.

**Comment by madsmtm** [comment]
It's because of the [branch protection rule](https://github.com/rust-windowing/winit/settings/branch_protection_rules/5589098); I intended to change that before I merged, but only just before

**Comment by kchibisov** [comment]
Ah, those are leftover tasks? We could merge and then change what is required.

**Comment by madsmtm** [comment]
Exactly

**Review by kchibisov — APPROVED** [review]

**Pull request merged**
